packages/core/ios/Sources/DdLogs.mm

```diff
@@ -0,0 +1,149 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+// Import this first to prevent require cycles
+#import <DatadogSDKReactNative-Swift.h>
+#import "DdLogs.h"
+
+
+@implementation DdLogs
+
+RCT_EXPORT_MODULE()
+
+RCT_REMAP_METHOD(debug, withDebugMessage:(NSString*)message
+                 withContext:(NSDictionary*)context
+                 withResolver:(RCTPromiseResolveBlock)resolve
+                 withRejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self debug:message context:context resolve:resolve reject:reject];
+}
```

Comment:
Do you need 2 layers here?
I think you can do this directly:
```suggestion
RCT_REMAP_METHOD(debug, withDebugMessage:(NSString*)message
                 withContext:(NSDictionary*)context
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)
{
    [self.ddLogsImplementation debugWithMessage:message context:context resolve:resolve reject:reject];
}
```

no?

Reply:
I will look into it!

Reply:
I looked into it and we can indeed build without having this duplication.
However, this duplication is what enables us to ensure the module matches the definition of the JS Spec, `RCT_REMAP_METHOD` does seem to include the method in the properties for now.

So it's a question of balance:
- Benefit: we get easy type checking when modifying an API + we are sure of the type checking
- Drawback: duplicate code whenever we add a new function

For me it's worth keeping it as it is now, RN might change in the future so we don't have to do this duplicate declaration and we can always remove it if it's too much pain to maintain in the end.